cpu/sam0_common/periph/eth.c

```diff
@@ -158,8 +158,13 @@ void sam0_eth_poweron(void)
 
     /* enable PHY */
     gpio_set(sam_gmac_config[0].rst_pin);
-    _is_sleeping = false;
 
+    /* if the PHY is not idle, it's likely broken */
+    if (!(GMAC->NSR.reg & GMAC_NSR_IDLE)) {
+        return;
```

Comment:
Maybe add a DEBUG statement here. That might help since the function doesn't return anything.
Would be interesting to know what happened to your PHY. But I guess it's out of the scope of this PR.